src/Aspire.Cli/Utils/PrerequisiteChecker.cs

```diff
@@ -0,0 +1,315 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Aspire.Cli.DotNet;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Cli.Utils;
+
+/// <summary>
+/// Result of prerequisite check.
+/// </summary>
+internal sealed class PrerequisiteCheckResult
+{
+    public bool IsValid => Errors.Count == 0;
+    public List<string> Warnings { get; init; } = [];
+    public List<string> Errors { get; init; } = [];
+}
+
+/// <summary>
+/// Service for checking system prerequisites.
+/// </summary>
+internal interface IPrerequisiteChecker
+{
+    /// <summary>
+    /// Checks all prerequisites for running Aspire CLI.
+    /// </summary>
+    Task<PrerequisiteCheckResult> CheckPrerequisitesAsync(CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Checks if .NET SDK version meets minimum requirements.
+    /// </summary>
+    Task<(bool IsValid, string? InstalledVersion, string? Message)> CheckDotNetSdkAsync(CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Checks if a container runtime (Docker/Podman) is available.
+    /// </summary>
+    Task<(bool IsAvailable, string? RuntimeName, string? Message)> CheckContainerRuntimeAsync(CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Checks if running in WSL and provides warnings if applicable.
+    /// </summary>
+    (bool IsWSL, string? Warning) CheckWSLEnvironment();
+
+    /// <summary>
+    /// Checks if Docker Engine (non-Desktop) is being used and may need tunnel configuration.
+    /// </summary>
+    Task<(bool IsDockerEngine, string? Message)> CheckDockerEngineAsync(CancellationToken cancellationToken = default);
+}
+
+internal sealed class PrerequisiteChecker : IPrerequisiteChecker
+{
+    private readonly IDotNetCliRunner _dotNetRunner;
+    private readonly ILogger<PrerequisiteChecker> _logger;
+    
+    // Minimum required .NET SDK version for Aspire CLI
+    // This should be kept in sync with the SDK version specified in global.json
+    private static readonly Version s_minimumDotNetVersion = new Version(10, 0);
```

Comment:
@mitchdenny @davidfowl is there a better place to have the minimum version defined?